generator/generator_base.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require 'erb'
+require 'pry'
+require 'fileutils'
+require 'net/http'
+require 'fhir_models'
+
+module Inferno
+  module Generator
+    class Base
+      def initialize(path, extras)
+        @path = path
+        @extras = extras
+        load_resources
+      end
+
+      def load_resources
+        @resource_by_path = Hash.new {}
+        @resources_by_type = Hash.new { |h, k| h[k] = [] }
+        Dir.glob("#{resource_file_path}/**/*.*") do |resource| # note one extra "*"
+          if File.file?(resource) && resource.end_with?('json', 'xml')
+
+            # We should consider using the native Ruby models instead of JSON
+            # There were problems with round-tripping certain SearchParameters though
```

Comment:
This should use the Ruby models instead of the json representation.  However, that would have been a pretty big lift on this PR, and I ran into some issues with SearchParamter extensions getting lost, and frankly the error message when you try to access a field that doesn't exist in a model produced tons of unuseful output.  I opened another ticket (FI-372)